### PR TITLE
fix(TDI-46848): MSCRM OAuth + on-premise mscrm lib

### PR DESCRIPTION
**Pull request by ypiel-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-46848

**What is the new behavior?**
client secret is now part of the oauth boday's form request.

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by olhava — DISMISSED** [review]

File: main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/src/main/java/org/talend/ms/crm/odata/ClientConfigurationFactory.java [hunk 1]
```diff
@@ -60,12 +60,13 @@ public final static ClientConfiguration buildNtlmClientConfiguration(String user
     }
 
     public final static ClientConfiguration buildOAuthPremiseClientConfiguration(String userName, String password, String authoryEndpoint,
-                                                                                 String serviceAPI, String clientId, String redirectUrl, String forcedResource) {
+                                                                                 String serviceAPI, String clientId, String clientSecret, String redirectUrl, String forcedResource) {
```
Comment (undx) [inline main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/src/main/java/org/talend/ms/crm/odata/ClientConfigurationFactory.java:63]: no final parameters

File: main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/src/main/java/org/talend/ms/crm/odata/authentication/OAuthPremiseStrategyImpl.java [hunk 2]
```diff
@@ -146,6 +146,9 @@ private String computeTokenUrl(String url){
         body.put("redirect_uri", conf.getRedirectURL());
         body.put("client_id", conf.getClientId());
 
+        if(conf.getClientSecret() != null && !"".equals(conf.getClientSecret().trim())){
```
Comment (undx) [inline main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/src/main/java/org/talend/ms/crm/odata/authentication/OAuthPremiseStrategyImpl.java:149]: space formatting around if clause.

**Review by undx — APPROVED** [review]
Some minor comments but LGTM as it's a backport

File: main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/src/main/java/org/talend/ms/crm/odata/ClientConfigurationFactory.java (hunk 1 above)
Comment (ypiel-talend) [reply main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/src/main/java/org/talend/ms/crm/odata/ClientConfigurationFactory.java:63]: Yes, but not related to the current issue.

File: main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/src/main/java/org/talend/ms/crm/odata/authentication/OAuthPremiseStrategyImpl.java (hunk 2 above)
Comment (ypiel-talend) [reply main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/src/main/java/org/talend/ms/crm/odata/authentication/OAuthPremiseStrategyImpl.java:149]: minor

**Review by olhava — APPROVED** [review]

**Pull request merged**